src/widget/input_box.c

```diff
@@ -5,8 +5,11 @@
 #include "graphics/text.h"
 #include "input/keyboard.h"
 
+static input_box *active_box;
+
 void input_box_start(input_box *box, uint8_t *text, int length, int allow_punctuation)
 {
+    active_box = box;
```

Comment:
Humm. I do need to find out if an input box is being used or not.

What do you think about making `input_box.c` own all input boxes (it would be a `static` variable) instead of each window having its own `input_box` (current implementation)?

Reply:
I migrated the code away from that so I'd not like going back 😉 The input boxes used to be lumped together but that was hard to maintain. I see the keyboard code is also not "safe" with respect to the `text` variable. I'll have a look at the whole setup to make it simpler and less error-prone.

Reply:
Thank you, that would be appreciated. My idea was to have Switch and Vita's virtual keyboard controlling each input box's text, and also only allowing the virtual keyboard to appear when there is an active input box.